source/vstd/arithmetic/div_mod.rs

```diff
@@ -289,72 +262,35 @@ pub proof fn lemma_div_pos_is_pos(x: int, d: int)
     lemma_div_induction_auto(d, x, |u: int| 0 <= u ==> u / d >= 0);
 }
 
-/// Proof that dividing a whole number by a natural number will result
-/// in a quotient that is greater than or equal to 0
-pub proof fn lemma_div_pos_is_pos_auto()
-    ensures
-        forall|x: int, d: int| 0 <= x && 0 < d ==> 0 <= #[trigger] (x / d),
-{
-    assert forall|x: int, d: int| 0 <= x && 0 < d implies 0 <= #[trigger] (x / d) by {
-        lemma_div_pos_is_pos(x, d);
-    }
-}
-
 /// Proof that dividing a number then adding 1 gives the same result
 /// as adding the divisor and then doing the division. Specifically,
 /// `1 + (x / d)` is equal to `(d + x) / d`.
-pub proof fn lemma_div_plus_one(x: int, d: int)
+pub broadcast proof fn lemma_div_plus_one(x: int, d: int)
     requires
         0 < d,
     ensures
+        #![trigger (1 + x / d), ((d + x) / d)]
         1 + x / d == (d + x) / d,
 {
     lemma_div_auto(d);
 }
 
-/// Proof that dividing a number then adding 1 gives the same result
-/// as adding the divisor and then doing the division
-pub proof fn lemma_div_plus_one_auto()
-    ensures
-        forall|x: int, d: int|
-            #![trigger (1 + x / d), ((d + x) / d)]
-            0 < d ==> 1 + (x / d) == (d + x) / d,
-{
-    assert forall|x: int, d: int| 0 < d implies #[trigger] (1 + x / d) == #[trigger] ((d + x)
-        / d) by {
-        lemma_div_plus_one(x, d);
-    }
-}
-
 /// Proof that dividing a number then subtracting 1 gives the same result
 /// as subtracting the divisor and then doing the division. Specifically,
 /// `-1 + (x / d)` is equal to `(-d + x) / d`.
-pub proof fn lemma_div_minus_one(x: int, d: int)
+pub broadcast proof fn lemma_div_minus_one(x: int, d: int)
     requires
         0 < d,
     ensures
+        #![trigger (-1 + x / d), ((-d + x) / d)]
         -1 + x / d == (-d + x) / d,
 {
     lemma_div_auto(d);
 }
 
-/// Proof that dividing a number then subtracting 1 gives the same result
-/// as subtracting the divisor and then doing the division
-pub proof fn lemma_div_minus_one_auto()
-    ensures
-        forall|x: int, d: int|
-            #![trigger (-1 + x / d), ((-d + x) / d)]
-            0 < d ==> -1 + x / d == (-d + x) / d,
-{
-    assert forall|x: int, d: int| 0 < d implies #[trigger] (-1 + x / d) == #[trigger] ((-d + x)
-        / d) by {
-        lemma_div_minus_one(x, d);
-    }
-}
-
 /// Proof that dividing any non-negative integer less than `d` by `d`
 /// produces a quotient of 0
-pub proof fn lemma_basic_div(d: int)
+pub proof fn lemma_basic_div_d(d: int)
```

Comment:
I wonder if there's a better name for this, like `lemma_basic_div_specific_divisor`?

Reply:
Changed to `lemma_basic_div_specific_divisor`, thanks!